Implement STM32 RTC alarms

markus-k opened the pull request:
Implements setting up and awaiting RTC alarms. 

Example:
```rust
    let mut rtc = Rtc::new(p.RTC, RtcConfig::default());

    rtc.set_alarm(
        Alarm::A,
        RtcAlarmMatch {
            subsecond: None,
            second: None,
            minute: None,
            hour: None,
            hour_is_pm: false,
            date: None,
        },
    );

    // would wait until the second increments
    rtc.wait_for_alarm(Alarm::A).await;
```

Still quite WIP as the different stm32 lines differ quite a bit in EXTI/IRQs. So far tested on STM32 G0, G4, L0 and F3.

Posting this here already to gather some feedback, if there is any. Maybe the actual implementation should be moved to the v2/v3 modules as they do differ quite a bit and need lots of conditionals.

Needs https://github.com/embassy-rs/stm32-data/pull/565 and https://github.com/embassy-rs/stm32-data/pull/564 merged and released in `stm32-data`.

markus-k committed:
Implement STM32 RTC alarms (WIP)

Implements setting up and awaiting RTC alarms. Still quite WIP
as the differnt stm32 lines differ quite a bit in EXTI/IRQs.

So far tested on STM32 G0, G4, L0 and F3.

Dirbaio commented:
> Needs https://github.com/embassy-rs/stm32-data/pull/565 and https://github.com/embassy-rs/stm32-data/pull/564 merged and released in stm32-data.

now that they're merged, switch embassy-stm32 to use stm32-data from git (uncomment the two lines, update the commit hash). We do stm32-metapac+embassy-stm32 releases at the same time.

markus-k committed:
Remove leftover logging statements in stm32 rtc alarm

markus-k committed:
Use latest stm32-metapac from git

markus-k committed:
Merge branch 'main' into stm32-rtc-alarms

markus-k committed:
Merge branch 'main' into stm32-rtc-alarms

markus-k committed:
Add RTC alarm example for stm32g0

markus-k committed:
Add rtc alarms to changelog

markus-k committed:
embassy-stm32: make rustfmt happy

markus-k committed:
Merge branch 'main' into stm32-rtc-alarms

markus-k committed:
embassy-stm32: Fixes for changes that happend upstream in the meanwhile

markus-k committed:
embassy-stm32: fix duplicate RTC implementation in rtc v2

markus-k committed:
embassy-stm32: remove alarm support for v2 rtc for now, implement for other v3 variants

implementing it for every single possible variant is just too much work for now

markus-k committed:
embassy-stm32: make rustfmt happy again

markus-k committed:
embassy-stm32: fix missing cfg for g0

leftger commented:
Hey @markus-k . This seems like a really useful PR. We recently implemented a new rule that all PRs have to be marked ready for review. I don't want this PR to go to waste. Could you address dirbaio's comment, fix the merge conflicts, make sure CI passes, and then mark it for review? You can even tag me and I'll take a gander :)

Dirbaio commented:
Hello,

The pull request backlog on the Embassy repo has grown uncontrollably. In an effort to clean it up, we're declaring "pull request bankruptcy" and closing all pull requests older than 1 year.

We've added a new [review queue](https://github.com/embassy-rs/embassy/blob/main/CONTRIBUTING.md) that works in a FIFO way, to avoid pull request reviews stalling for so long in the future.

If you're still interested in contributing this PR, you're welcome to open a new one. Please rebase on top of latest main branch if you do so the changes merge cleanly.

Thank you for your understanding!
